crates/nu-protocol/src/engine/engine_state.rs

```diff
@@ -957,12 +947,59 @@ impl EngineState {
             .map(|comment_spans| self.build_usage(comment_spans))
     }
 
+    /// Returns the current working directory, which is guaranteed to be canonicalized.
+    ///
+    /// Returns an empty String if $env.PWD doesn't exist.
+    #[deprecated(since = "0.92.3", note = "please use `EngineState::cwd()` instead")]
     pub fn current_work_dir(&self) -> String {
-        self.get_env_var("PWD")
-            .map(|d| d.coerce_string().unwrap_or_default())
+        self.cwd(None)
+            .map(|path| path.to_string_lossy().to_string())
             .unwrap_or_default()
     }
 
+    /// Returns the current working directory, which is guaranteed to be an absolute path
+    /// but might contain symlink components.
+    ///
+    /// If `stack` is supplied, also considers modifications to the working directory on the stack
+    /// that have yet to be merged into the engine state.
+    ///
+    /// Returns an error if $env.PWD doesn't exist, is not a String, or is not an absolute path.
+    pub fn cwd(&self, stack: Option<&Stack>) -> Result<PathBuf, ShellError> {
```

Comment:
I think it would be better to return a String which users can convert to PathBuf if they need to. Otherwise, there will be unnecessary PathBuf->String conversion if string is required.

Reply:
I'm not sure. Conceptually, "current working directory" should be a path, not a string. It also matches the behavior of `std::env::current_dir()`. Zero-cost conversion from `PathBuf` to `String` can be done with `path_buf.into_os_string().into_string().unwrap()`, but that involves an `unwrap()`.

On the other hand, returning a string does make some use cases more convenient. Maybe I'll add a `cwd_as_string()` method? Just one more API won't hurt...?

Reply:
Yeah, one more API call is fine, that's why I made `env::current_dir()` and `env::current_dir_str()`. I'd try to avoid the `unwrap()`.

Reply:
I think I'll add the API later while migrating the old API usages. It should give me a better idea whether an extra API is needed.